From 6ea74a67ef916624fe02d52305cd2d524057a771 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 7 Jun 2022 17:41:45 +0800
Subject: [PATCH 2982/4585] [Backport] sw64: numa: switch to arch node_distance

Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56TYY

Reference: https://gitee.com/openeuler/kernel/commit/829abaa21e9770da988330d09aac2d4c446e1887

--------------------------------

Switch to arch node_distance to read distance from node distance table
to get a more accurate node distance.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>

Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 arch/sw_64/include/asm/topology.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/sw_64/include/asm/topology.h b/arch/sw_64/include/asm/topology.h
index 9f45b550abca..f8242d00290b 100644
--- a/arch/sw_64/include/asm/topology.h
+++ b/arch/sw_64/include/asm/topology.h
@@ -45,6 +45,8 @@ extern const struct cpumask *cpumask_of_node(int node);
 extern void numa_add_cpu(unsigned int cpu);
 extern void numa_remove_cpu(unsigned int cpu);
 extern void numa_store_cpu_info(unsigned int cpu);
+extern int __node_distance(int from, int to);
+#define node_distance(a, b) __node_distance(a, b)
 #define parent_node(node) (node)
 #define cpumask_of_pcibus(bus)	(cpu_online_mask)
 #else /* !CONFIG_NUMA */
-- 
2.17.1

